temporalio/worker/workflow_instance.py

```diff
@@ -0,0 +1,1655 @@
+                            ] = self.create_future()
+
+        # Create the handle and set as pending
+        handle = _ChildWorkflowHandle(
+            self, seq, input, start_fut, result_fut, run_child()
+        )
+        self._pending_child_workflows[seq] = handle
+        handle._apply_start_command(self._add_command())
+
+        # Wait on start
+        await start_fut
+        return handle
+
+    #### Miscellaneous helpers ####
+    # These are in alphabetical order.
+
+    def _add_command(self) -> temporalio.bridge.proto.workflow_commands.WorkflowCommand:
+        return self._current_completion.successful.commands.add()
+
+    async def _cancel_external_workflow(
+        self,
+        # Should not have seq set
+        command: temporalio.bridge.proto.workflow_commands.WorkflowCommand,
+    ) -> None:
+        seq = self._next_seq("external_cancel")
+        done_fut = self.create_future()
+        command.request_cancel_external_workflow_execution.seq = seq
+
+        # Set as pending
+        self._pending_external_cancels[seq] = done_fut
+
+        # Wait until done (there is no cancelling a cancel request)
+        await done_fut
+
+    def _check_condition(self, fn: Callable[[], bool], fut: asyncio.Future) -> bool:
+        if fn():
+            fut.set_result(True)
+            return True
+        return False
+
+    def _convert_payloads(
+        self,
+        payloads: Sequence[temporalio.bridge.proto.common.Payload],
+        types: Optional[List[Type]],
+    ) -> List[Any]:
+        if not payloads:
+            return []
+        # Only use type hints if they match count
+        if types and len(types) != len(payloads):
+            types = None
+        try:
+            return self._payload_converter.from_payloads(
+                temporalio.bridge.worker.from_bridge_payloads(payloads),
+                type_hints=types,
+            )
+        except Exception as err:
+            raise RuntimeError("Failed decoding arguments") from err
+
+    def _next_seq(self, type: str) -> int:
+        seq = self._curr_seqs.get(type, 0) + 1
+        self._curr_seqs[type] = seq
+        return seq
+
+    def _run_once(self) -> None:
+        try:
+            asyncio._set_running_loop(self)
+
+            # We instantiate the workflow class _inside_ here because __init__
+            # needs to run with this event loop set
+            if not self._object:
+                self._object = self._defn.cls()
+
+            # Run while there is anything ready
+            while self._ready and not self._force_stop_loop:
+
+                # Run and remove all ready ones
+                while self._ready and not self._force_stop_loop:
+                    handle = self._ready.popleft()
+                    handle._run()
+
+                # Check conditions which may add to the ready list
+                self._conditions[:] = [
+                    t for t in self._conditions if not self._check_condition(t[0], t[1])
```

Comment:
nit: would be more readable as:
```suggestion
                    (callback, future) for (callback, future) in self._conditions if not self._check_condition(callback, future)
```

Reply:
Less readable to me, at that point I'd make a for loop instead of a comprehension if I to destructure a tuple, reference it, and restructure it again

Reply:
Fine, could also do `*t` instead..

Reply:
I'll confirm MyPy properly checks that, then will do

Reply:
Updated